env/io_posix.cc

```diff
@@ -56,6 +56,65 @@ int Fadvise(int fd, off_t offset, size_t len, int advice) {
 #endif
 }
 
+// On MacOS (and probably *BSD), the posix write and pwrite calls do not support
+// buffers larger than 2^31-1 bytes. These two wrappers fix this issue by
+// cutting the buffer in 1GB chunks when having to write buffers larger than
+// 2GB. We use this chunk size to be sure to keep the writes aligned.
+
+bool PosixWriteWrapper(int fd, const void* buf, size_t nbyte) {
```

Comment:
"Wrapper" is not very clear. What if we just call it Write() or something like that.

Also, why `const void*`? Why not directly pass `const char*`?

Reply:
Using `const void*` instead of `const char*` was just to mimic the posix call. But I have no strong argument for this choice. Will change.